### PR TITLE
Fix incorrect advance when decoding v1

### PR DESCRIPTION
**Pull request by Goirad**
This fixes a bug in decoding version 1 headers that leaves the buffer
pointing to the last newline instead of consuming it. Also adds tests to
confirm the fix.

### CODE REVIEW

File: src/protocol/mod.rs [hunk 1]
```diff
@@ -292,6 +292,7 @@ impl ProxyHeader {
             buf.advance(1);
             return Err(Version1ParsingError::ExpectedCRLF.into());
         }
+        buf.advance(1);
```
Comment (Proximyst) [inline src/protocol/mod.rs]: This looks sane, however perhaps it'd be better to just `get_u8` the byte and instead compare that? That way we advance 1 no matter what (which we do anyways), and avoid an unnecessary slice.

**Review by Proximyst — CHANGES_REQUESTED** [review]
Thanks for your contribution! Small note, and it'll be good to go :)

This'll be published on a minor version bump, as although it is a bug, it is also a behavioural difference.

File: src/protocol/mod.rs (hunk 1 above)
Comment (Goirad) [reply src/protocol/mod.rs]: Done, great suggestion

**Review by Proximyst — APPROVED** [review]
Thanks!

**Pull request merged**